src/test/java/com/gravity9/mongocdc/ChangeStreamTest.java

```diff
@@ -19,100 +21,174 @@
+    }
+
+    @Test
+    void shouldNotHandleEventsWhenListenerIsDeregisteredFromAllPartitions() throws Exception {
+        MongoCDCManager manager = new MongoCDCManager(mongoConfig);
+
+        TestChangeStreamListener listener0 = new TestChangeStreamListener();
+        manager.registerListenerToAllPartitions(listener0);
+        manager.start();
+
+        insertDocumentsToAllPartitions();
+
+        // Wait for CDC event to arrive
+        Thread.sleep(500);
+
+        List<ChangeStreamDocument<Document>> events0 = listener0.getEvents();
+        assertEquals(3, events0.size());
+
+        // when
+        manager.deregisterListenerFromAllPartitions(listener0);
+
+        // then
+        var deleteResult0 = collection.deleteOne(Filters.eq("_id", new ObjectId(TestIds.MOD_0_ID)));
+        var deleteResult1 = collection.deleteOne(Filters.eq("_id", new ObjectId(TestIds.MOD_1_ID)));
+        var deleteResult2 = collection.deleteOne(Filters.eq("_id", new ObjectId(TestIds.MOD_2_ID)));
+        assertEquals(1, deleteResult0.getDeletedCount());
+        assertEquals(1, deleteResult1.getDeletedCount());
+        assertEquals(1, deleteResult2.getDeletedCount());
+
+        // Wait for CDC event to arrive
+        Thread.sleep(500);
+
+        // only old events are available
+        List<ChangeStreamDocument<Document>> events1 = listener0.getEvents();
+        assertEquals(3, events1.size());
+    }
+
+    @Test
+    void shouldNotHandleEventsFromSelectedPartitionWhenListenerIsDeregisteredFromThisPartition() throws Exception {
+        MongoCDCManager manager = new MongoCDCManager(mongoConfig);
+
+        TestChangeStreamListener listener0 = new TestChangeStreamListener();
+        manager.registerListenerToAllPartitions(listener0);
+        manager.start();
+
+        insertDocumentsToAllPartitions();
+
+        // Wait for CDC event to arrive
+        Thread.sleep(500);
+
+        List<ChangeStreamDocument<Document>> events0 = listener0.getEvents();
+        assertEquals(3, events0.size());
+
+        // when
+        manager.deregisterListener(listener0, List.of(0));
+
+        // then
+        var deleteResult0 = collection.deleteOne(Filters.eq("_id", new ObjectId(TestIds.MOD_0_ID)));
+        var deleteResult1 = collection.deleteOne(Filters.eq("_id", new ObjectId(TestIds.MOD_1_ID)));
+        var deleteResult2 = collection.deleteOne(Filters.eq("_id", new ObjectId(TestIds.MOD_2_ID)));
+        assertEquals(1, deleteResult0.getDeletedCount());
+        assertEquals(1, deleteResult1.getDeletedCount());
+        assertEquals(1, deleteResult2.getDeletedCount());
+
+        // Wait for CDC event to arrive
+        Thread.sleep(500);
+
+        // only old events and deletes from not deregistered partitions are available
+        List<ChangeStreamDocument<Document>> events1 = listener0.getEvents();
+        assertEquals(5, events1.size());
+    }
+
+    private void insertDocumentsToAllPartitions() {
```

Comment:
Should return number of events generated and the value should be used in asserts.

Reply:
done